Modules/documents/Controller/DocumentslistController.php

```diff
@@ -62,6 +143,30 @@ public function editAction($id_space, $id) {
         $form->setTitle(DocumentsTranslator::Edit_Document($lang));
         $form->addText("title", DocumentsTranslator::Title($lang), true, $data["title"]);
         $form->addUpload("file_url", DocumentsTranslator::File($lang));
+
+
+        $plan = new CorePlan($this->currentSpace['plan'], $this->currentSpace['plan_expire']);
+        if($plan->hasFlag(CorePlan::FLAGS_DOCUMENTS)) {
+            $form->addSelectMandatory(
+                "visibility",
+                "Visibility",
+                ["Private", "Members", "Protected User", "Protected Client", "Public"],
+                [Document::$VISIBILITY_PRIVATE, Document::$VISIBILITY_MEMBERS, Document::$VISIBILITY_USER, Document::$VISIBILITY_CLIENT, Document::$VISIBILITY_PUBLIC],
+                $data['visibility'] ?? Document::$VISIBILITY_PRIVATE
+            );
+            $mu = new CoreUser();
+            $users = $mu->getSpaceActiveUsersForSelect($id_space, "name");
+            $mc = new ClClient();
+            $clients = $mc->getForList($id_space);
+            $clients['names'] = array_merge([''], $clients['names']);
+            $clients['ids'] = array_merge([0], $clients['ids']);
+            $form->addSelect('id_ref_user', CoreTranslator::User($lang), $users['names'], $users['ids'], $data['id_ref'] ?? 0);
+            $form->addSelect('id_ref_client', ClientsTranslator::Client($lang), $clients['names'], $clients['ids'], $data['id_ref'] ?? 0);
```

Comment:
An ideal would be to display only one (or none) of these two inputs based on visibility choice. It is possible to use DynamicForms.js to do that dynamically (but need to add a new method/option for displaying / hiding inputs)